src/ol/renderer/webgl/PointsLayer.js

```diff
@@ -296,20 +297,22 @@ class WebGLPointsLayerRenderer extends WebGLLayerRenderer {
     const layer = this.getLayer();
     const vectorSource = layer.getSource();
     const viewState = frameState.viewState;
-
-    // the source has changed: clear the feature cache & reload features
+    const viewNotMoving = !frameState.viewHints[ViewHint.ANIMATING] && !frameState.viewHints[ViewHint.INTERACTING];
+    const extentChanged = !equals(this.previousExtent_, frameState.extent);
     const sourceChanged = this.sourceRevision_ < vectorSource.getRevision();
+
     if (sourceChanged) {
       this.sourceRevision_ = vectorSource.getRevision();
+    }
 
+    if (viewNotMoving && (extentChanged || sourceChanged)) {
       const projection = viewState.projection;
       const resolution = viewState.resolution;
-      vectorSource.loadFeatures([-Infinity, -Infinity, Infinity, Infinity], resolution, projection);
-    }
 
-    const viewNotMoving = !frameState.viewHints[ViewHint.ANIMATING] && !frameState.viewHints[ViewHint.INTERACTING];
-    const extentChanged = !equals(this.previousExtent_, frameState.extent);
-    if ((sourceChanged || extentChanged) && viewNotMoving) {
+      const renderBuffer = layer instanceof BaseVector ? layer.getRenderBuffer() : 0;
```

Comment:
did you use `instanceof` because of an error from the type checker?

Reply:
Actually the renderer can use layers that do not inherit from the `BaseVector` class, as seen here: https://github.com/openlayers/openlayers/blob/master/src/ol/layer/WebGLPoints.js

The only requirement is having a vector source.

Reply:
ok, got it now.